### PR TITLE
Add CalculatedSummary block type

### PR DESCRIPTION
**Pull request by dwyeradam**
Json schema checks
Answers used in calculation must be of same type (answer type, unit, currency)

### CODE REVIEW

File: app/validation/validator.py [hunk 1]
```diff
@@ -204,6 +207,38 @@ def validate_repeat_rule_restricted(rule, block):
...
         return errors
 
+
+    def validate_calculated_summary_type(self, block, answers_with_parent_ids):
+        answer_types = [
+            answers_with_parent_ids[answer_id]['answer']['type']
+            for answer_id in block['calculation']['answers_to_calculate']
+        ]
+
+        if not all(type == answer_types[0] for type in answer_types):
```
Comment (benhowes) [inline app/validation/validator.py]: Would be best not to override the builtin `type` variable here and line 227, 236

File: schemas/blocks/calculated_summary.json [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+        "properties": {
+          "calculation_type": {
+            "type": "string",
+            "enum": [
+              "sum"
+            ]
+          },
+          "answers_to_calculate": {
+            "type": "array",
+            "minItems": 2,
```
Comment (benhowes) [inline schemas/blocks/calculated_summary.json]: In the scenario that you're testing a survey part way through building it, is it conceivable that you'd run a survey where you've used a calculate block, but only have a single answer id in this block at the point you want to run a test of your other logic? My point is that it's probably a bit silly to only have one item in `answers_to_calculate`, but perhaps it shouldn't be invalid?

File: app/validation/validator.py (hunk 1 above)
Comment (dwyeradam) [reply app/validation/validator.py]: Yes, stupid of me, will call it answer_type.

File: schemas/blocks/calculated_summary.json (hunk 2 above)
Comment (dwyeradam) [reply schemas/blocks/calculated_summary.json]: I agree in that situation you may want just one answer but I believe schema validator is to protect against poorly designed schemas and to do that we have to treat each survey as complete at the point of validation. If you only need one answer then you should use the Confirmation block type and change it after.

Comment (benhowes) [reply schemas/blocks/calculated_summary.json]: Okay @dwyeradam, I'll back off on that. I do wonder if having the ability to raise "Warnings/Recommendations" as well as hard errors may be useful in future?

Comment (dwyeradam) [reply schemas/blocks/calculated_summary.json]: @benhowes Only back off if you're happy with my response. Otherwise we open it up to the group and hope for some consensus. Often with varying success. I won't take it personally.

**Comment by MebinAbraham** [comment]
Ideally would like to test against the runner changes.

**Comment by dwyeradam** [comment]
@benhowes did you want a quick look at the latest two commits. Just two additional tests. One to check for invalid answer_ids and another to test for duplicate answer ids.

**Comment by benhowes** [comment]
@dwyeradam Looks good to me! I should've thought of the duplicate id's thing too!
